### PR TITLE
Detect SPM usage, add SPM cache steps

### PR DESCRIPTION
**Pull request by ofalvai**
### Changes

- Detect SPM usage in the project scanning phase
- Add SPM cache steps in the workflow generation phase

### CODE REVIEW

**Comment by ofalvai** [comment]
Current dependencies on/for this PR:
* master
  * **PR #250** <a href="https://app.graphite.dev/github/pr/bitrise-io/bitrise-init/250" target="_blank">Graphite</a> 
    * **PR #251** <a href="https://app.graphite.dev/github/pr/bitrise-io/bitrise-init/251" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/bitrise-io/bitrise-init/251?utm_source=stack-comment).

File: scanners/ios/spm.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	if len(matches) > 0 {
+		return true, nil
+	}
+
+	// Xcode projects: find lockfile inside `xcodeproj`
+	xcodeFilters := []pathutil.FilterFunc{
+		pathutil.BaseFilter("Package.resolved", true), // match nested project folders too
+		pathfilters.ForbidPodsDirComponentFilter,      // don't match dependency source checkouts
```
Comment (lpusok) [inline scanners/ios/spm.go:29]: Could filter out node_modules folder (react projects could have it)

Comment (ofalvai) [reply scanners/ios/spm.go:29]: But is it possible for an NPM package to contain Swift code?

Comment (lpusok) [reply scanners/ios/spm.go:29]: Theoretically possible that it contains an Xcode project, I did seen something like this.

Comment (ofalvai) [reply scanners/ios/spm.go:30]: Fixed!

**Review by lpusok — DISMISSED** [review]

**Comment by ofalvai** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: 6dQCJ6udRdY0ZfxVWVla.
See details on [graphite.dev](https://app.graphite.dev/github/pr/bitrise-io/bitrise-init/251)

**Review by lpusok — APPROVED** [review]

**Pull request merged**